src/components/EtherspotTransactionKit.tsx

```diff
@@ -10,34 +11,67 @@ import ProviderWalletContextProvider from '../providers/ProviderWalletContextPro
 import EtherspotContextProvider from '../providers/EtherspotContextProvider';
 
 interface EtherspotTransactionKitProps extends React.PropsWithChildren {
-  provider: WalletProviderLike;
+  provider: WalletProviderLike | WalletProviderLikeModular;
   chainId?: number;
   accountTemplate?: AccountTemplate;
   dataApiKey?: string;
   bundlerApiKey?: string;
+  modular?: boolean;
```

Comment:
This is the part we definitely have to change - we can't let developers think that every accountTemplate `could` or `couldn't` be modular